src/AdScreenerTool.js

```diff
@@ -11,14 +13,16 @@ import AdUnit from "./AdUnit.js";
 import TimePeriodPicker from "./TimePeriodPicker.js";
 import FilterSelector from "./FilterSelector.js";
 import LabelEntryForm from "./LabelEntryForm.js";
+import SearchField from "./SearchField.js";
 
 const getAdsURL = "/getads";
 const getFilterSelectorDataURL = "/filter-options";
 
 const disableOptions = false;
 
-function getSelectorValue(array,param){
-  return (param===undefined) ? array[0] : array[array.findIndex(element => element.value === param)]
+function getSelectorValue(array,param,topicFlag){
+  var defaultValue = topicFlag ? "" : array[0];
+  return (param===undefined) ? defaultValue : array[array.findIndex(element => element.value === param)]
```

Comment:
instead of handling topic specifically in this general function can this be handled only in handleSearch function? ie only clearing the value of topic when full text search is selected/used?